cmd/manager/main.go

```diff
@@ -303,3 +314,46 @@ func initCustomZapLogger(level, encoding string) (*zap.Logger, error) {
 	}
 	return cfg.Build()
 }
+
+func configureDeletionProtectionFlags(config *Config) {
+	if config == nil {
+		return
+	}
+
+	objectDeletionSet := false
+	subObjectDeletionSet := false
+
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == "object-deletion-protection" {
+			objectDeletionSet = true
+		}
+
+		if f.Name == "subobject-deletion-protection" {
+			subObjectDeletionSet = true
+		}
+	})
+
+	if !objectDeletionSet {
+		objDeletion := os.Getenv("OBJECT_DELETION_PROTECTION")
```

Comment:
Nit
Just curious: should we be case insensitive here?
```suggestion
		objDeletion := strings.ToLower(os.Getenv("OBJECT_DELETION_PROTECTION"))
```